rclpy/rclpy/validate_full_topic_name.py

```diff
@@ -12,12 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from typing import Literal
+
 from rclpy.exceptions import InvalidServiceNameException
 from rclpy.exceptions import InvalidTopicNameException
 from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
 
 
-def validate_full_topic_name(name, *, is_service=False):
+def validate_full_topic_name(name: str, *, is_service: bool = False) -> Literal[True]:
```

Comment:
Should we remove the typing info from the param documentation below?

```diff
-:param name str: topic or service name to be validated
+:param name: topic or service name to be validated
```

same for all other changes.

Reply:
@InvincibleRMC i think we want to remove the typing information from the doc part if that is typed. **i believe that we should avoid inconsistent typing information as documentation, so if typing is available, probably we do not need to have types in doc section. typing is the correct one.**

if typing is available, it shows on API arguments.

![image](https://github.com/ros2/rclpy/assets/43395114/c194cfdc-0c57-44f9-8fbc-873853af04cb)

on the other hand, doc section comes up like this as an example.

![image](https://github.com/ros2/rclpy/assets/43395114/236b73a2-3ff4-4c4a-9811-4839f73c6f52)

Reply:
That makes sense to me.